sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/WebJobsServiceBusTestBase.cs

```diff
@@ -0,0 +1,195 @@
+using Microsoft.Extensions.Hosting;
+using NUnit.Framework;
+using static Azure.Messaging.ServiceBus.Tests.ServiceBusScope;
+
+namespace Microsoft.Azure.WebJobs.Host.EndToEndTests
+{
+    [NonParallelizable]
+    [LiveOnly]
+    public class WebJobsServiceBusTestBase
+    {
+        // surrounding with % indicates that this is used as a pointer to an app setting rather than
+        // the literal value of the queue/topic/etc
+        private const string _firstQueueNameKey = "webjobstestqueue1";
+        protected const string FirstQueueNameKey = "%" + _firstQueueNameKey + "%";
+
+        private const string _secondQueueNameKey = "webjobstestqueue2";
+        protected const string SecondQueueNameKey = "%" + _secondQueueNameKey + "%";
+
+        private const string _thirdQueueNameKey = "webjobstestqueue3";
+        protected const string ThirdQueueNameKey = "%" + _thirdQueueNameKey + "%";
+
+        private const string _topicNameKey = "webjobstesttopic";
+        protected const string TopicNameKey = "%" + _topicNameKey + "%";
+
+        private const string _firstSubscriptionNameKey = "webjobstestsubscription1";
+        protected const string FirstSubscriptionNameKey = "%" + _firstSubscriptionNameKey + "%";
+
+        private const string _secondSubscriptionNameKey = "webjobstestsubscription2";
+        protected const string SecondSubscriptionNameKey = "%" + _secondSubscriptionNameKey  + "%";
+
+        private const string _secondaryNamespaceQueueKey = "webjobtestsecondarynamespacequeue";
+        protected const string SecondaryNamespaceQueueNameKey = "%" + _secondaryNamespaceQueueKey + "%";
+
+        // the connection key shouldn't use the % because the value specified is assumed to be a pointer
+        protected const string SecondaryConnectionStringKey = "webjobtestsecondaryconnection";
+
+        protected const int SBTimeoutMills = 120 * 1000;
+        protected const int DrainWaitTimeoutMills = 120 * 1000;
+        protected const int DrainSleepMills = 5 * 1000;
+        internal const int MaxAutoRenewDurationMin = 5;
+        internal static TimeSpan HostShutdownTimeout = TimeSpan.FromSeconds(120);
+
+        protected static QueueScope _firstQueueScope;
+        protected static QueueScope _secondaryNamespaceQueueScope;
+        private QueueScope _secondQueueScope;
+        private QueueScope _thirdQueueScope;
+        protected static TopicScope _topicScope;
+
+        private readonly bool _isSession;
+
+        protected WebJobsServiceBusTestBase(bool isSession)
+        {
+            _isSession = isSession;
+        }
+
+        /// <summary>
+        ///   Performs the tasks needed to initialize the test fixture.  This
+        ///   method runs once for the entire fixture, prior to running any tests.
+        /// </summary>
+        ///
+        [SetUp]
+        public async Task FixtureSetUp()
+        {
+            _firstQueueScope = await CreateWithQueue(enablePartitioning: false, enableSession: _isSession);
```

Comment:
Do we need 3 scopes for every test? This might make tests slow.

Reply:
Not for every test, but I didn't want to spend a lot of time optimizing until after we do the migration.